uvicorn/protocols/websockets/wsproto_impl.py

```diff
@@ -249,14 +252,15 @@ async def run_asgi(self) -> None:
             result = await self.app(self.scope, self.receive, self.send)
         except BaseException:
             self.logger.exception("Exception in ASGI application\n")
-            if not self.handshake_complete:
+            if not self.response_started:
                 self.send_500_response()
             self.transport.close()
         else:
             if not self.handshake_complete:
                 msg = "ASGI callable returned without completing handshake."
                 self.logger.error(msg)
-                self.send_500_response()
+                if not self.response_started:
+                    self.send_500_response()
```

Comment:
Why are those necessary?

What's the behavior if they are not present?

Reply:
This is similar to the code in http/h11_impl.py, e.g.
```python
        except BaseException as exc:
            msg = "Exception in ASGI application\n"
            self.logger.error(msg, exc_info=exc)
            if not self.response_started:
                await self.send_500_response()
            else:
                self.transport.close()
```
If an attempt is made to send a 500_response() after the sending of a response has already begun, then the application fails in a nasty way.  send_500_response() fails in a way which isn't caught, no error is output or anything.  This is what was causing me to comment this:

```
I am unable to silence the wsproto internal error which happens when it is trying to send a 500 error.

wsproto will start a response as soon as it gets the .start message
It then gets an invalid message and attempts to send a 500 response
but it cannot because it has already started a response.
An exception is not raised via the async send() method but somewhere else which I cannot fully understand, so the test fails.
I've added a skip for the WSProto case, but maybe there is a better solution? Skipping is not ideal, because this test was designed to increase coverage.
```

We could of course try to fix that error, it is bad if an internal exception is hard to debug, but that's not within the scope of this PR.  
I guess the check could be moved into `send_500_response()` but I was following the style of `h11_impl.py`

Reply:
If you approve, I´ll move this check into `send_500_response()` where it properly belongs, along with a comment, and do the same in h11_impl.py for consistency.  We can also log the situation.

Reply:
still something broken with the GHA.  I'll leave this until we sort out what's going on.